modules/ding_ipe_filter/ding_ipe_filter.module

```diff
@@ -0,0 +1,146 @@
+    'type' => MENU_DEFAULT_LOCAL_TASK,
+    'weight' => 1,
+  );
+  $items['admin/config/user-interface/ipe-filter/panes'] = array(
+    'title' => 'IPE pane filter',
+    'description' => 'Choose what content should be displayed on the add content modal page.',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('ding_ipe_filter_admin_panes_form'),
+    'access arguments' => array('access administration pages'),
+    'file' => 'ding_ipe_filter.admin.inc',
+    'type' => MENU_LOCAL_TASK,
+    'weight' => 2,
+  );
+
+  return $items;
+}
+
+/**
+ * Implements hook_theme().
+ */
+function ding_ipe_filter_theme() {
+  return array(
+    'ding_ipe_filter_table' => array(
+      'render element' => 'form',
+    ),
+  );
+}
+
+/**
+ * Implements theme_hook().
+ */
+function theme_ding_ipe_filter_table($variables) {
+  $form = $variables['form'];
+  $header = $form['#header'];
+
+  $content = array(
+    'header' => $header,
+    'rows' => array(),
+  );
+  foreach (element_children($form) as $row_index) {
+    if (isset($form[$row_index]['title'])) {
+      $content['rows'][] = array(
+        drupal_render($form[$row_index]['title']),
+        drupal_render($form[$row_index]['value']),
+      );
+    }
+    else {
+      $content['rows'][] = array(array(
+        'data' => drupal_render($form[$row_index]),
+          'colspan' => 2,
+          'header' => TRUE,
+        ),
+      );
+    }
+  }
+  $output = theme('table', $content);
+  $output .= drupal_render_children($form);
+  return $output;
+}
+
+/**
+ * Implements hook_theme_registry_alter().
+ */
+function ding_ipe_filter_theme_registry_alter(&$registry) {
+  global $user;
+  $ipe_roles = variable_get('ding_ipe_filter_roles', array());
+
+  if (!empty($user->roles)) {
+    $role_names = array_values($user->roles);
+    $user_roles = array_combine($role_names, $role_names);
+    $roles = array_intersect(array_keys($user_roles), $ipe_roles);
+  }
+  if (!empty($roles)) {
+    $registry['panels_ipe_add_pane_button']['function'] = '_ding_ipe_filter_add_pane_buttons';
+  }
+}
+
+/**
+ * Add pane buttons on IPE filter menu based on settings and enabled modules.
+ *
+ * @param array $variables
+ *   Theme registry variables.
+ *
+ * @return string
+ *   Rendered IPE filter menu.
+ */
+function _ding_ipe_filter_add_pane_buttons($variables = array()) {
+  ctools_include('content');
+  $content_types = ctools_content_get_available_types();
+  $panel_editor = new panels_renderer_editor();
+  $categories = $panel_editor->get_categories($content_types);
+
+  $assets_loaded = &drupal_static('ipe_assets_loaded', FALSE);
+  $settings = variable_get('ding_ipe_filter_table', array());
+  $region_id  = $variables['region_id'];
+  $renderer = $variables['renderer'];
+
+  $buttons = array();
+  $menu_width = 'calc(100% - 18px)';
+  $attributes = array(
+    'class' => 'ctools-use-modal ding-ipe-menu-item',
+    'style' => 'width: ' . $menu_width . '; height: auto;',
```

Comment:
My point is: The module includes `ipe.css` and I would expect it to be set there. If there is a specific reason to set it here instead of in the CSS file then I would like a code comment explaining why.